### PR TITLE
add `bitround_along_dim` for Fig3

### LINKED ISSUE #16: Port plotting functions to `python`
Overview: 
- [x] Fig. 2 #19 
- [ ] Fig. 3 seems to work nice with an `xr_bitround` wrapper 
- [x] Fig. SI 1: #59 

--- 
The plotting function gives a great overview about the information content present in each bit of a variable.
The port of the function to python should be straight forward and should be preferred over a further call of to Julia.

### PR DESCRIPTION
**Pull request by aaronspring**
Closes #16

- `bitround_along_dim` create xr.DataArray/xr.Dataset for this plot:
image
- `add_labels_fig3` adds the lines and labels

### CODE REVIEW

**Comment by aaronspring** [comment]
probably it would be better when allowing another keyword: `keepbits=[]` which could be used instead of `inflevels=[]`, Fig 3 Kloewer use keepbits I guess. with inflevels, you may end up with the same bitrounding for different levels.

File: bitinformation_pipeline/graphics.py [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+    """
+    Helper function for visualization of Figure 3 in Klöwer et al. 2021.
+
+    Klöwer, M., Razinger, M., Dominguez, J. J., Düben, P. D., & Palmer, T. N. (2021).
+    Compressing atmospheric data into its real information content. Nature Computational Science, 1(11), 713–724. doi: 10/gnm4jj
+
+    Inputs
+    ------
+    ds : xr.DataArray
+      plotted before
```
Comment (observingClouds) [inline bitinformation_pipeline/graphics.py:29]:
```suggestion
      Plotted dataset
```

File: bitinformation_pipeline/graphics.py [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+      distance between `keepbits = int` and `x%` label. Defaults to 8.
+    kwargs : dict
+      kwargs to be passed to `ax.text` and `ax.plot`. Use `transform=ccrs.Geodetic()` when using `cartopy`
+
+    Returns
+    -------
+
+    Example
+    -------
+    Plotting an single-dimension coordinate dataset:
```
Comment (observingClouds) [inline bitinformation_pipeline/graphics.py:56]:
```suggestion
    Plotting a single-dimension coordinate dataset:
```

File: bitinformation_pipeline/graphics.py [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+    >>> inflevels = [1.0, 0.9999, 0.99, 0.975, 0.95]
+    >>> ds_bitrounded_along_lon = bp.bitround.bitround_along_dim(
+    ...     ds, info_per_bit, dim="lon", inflevels=inflevels
+    ... )
+    >>> diff = (ds - ds_bitrounded_along_lon)["air"].isel(time=0)
+    >>> diff.plot()  # doctest: +ELLIPSIS
+    <matplotlib.collections.QuadMesh object at ...>
+    >>> add_labels_fig3(diff, info_per_bit, inflevels)  # doctest: +ELLIPSIS
+
+    Plotting an multi-dimensional coordinate dataset:
```
Comment (observingClouds) [inline bitinformation_pipeline/graphics.py:68]:
```suggestion
    Plotting a multi-dimensional coordinate dataset
```

**Review by observingClouds — CHANGES_REQUESTED** [review]
Could we make the usage/names of `add_labels_fig3` a bit clearer? How about calling it `add_bitinfo_labels`?

**Pull request merged**